athenz/utils.go

```diff
@@ -401,3 +443,89 @@ func validateDatePatternFunc(validPattern string, attribute string) schema.Schem
 		return nil
 	}
 }
+
+func validateRoleMember(members []interface{}, settings map[string]interface{}) error {
+	for _, mRaw := range members {
+		data := mRaw.(map[string]interface{})
+		name := data["name"].(string)
+
+		expirationDays := 0
+		reviewDays := 0
+		memberType := ""
+
+		if strings.HasPrefix(name, "user.") {
+			memberType = "user"
+			if settings["user_expiry_days"] != nil {
+				expirationDays = settings["user_expiry_days"].(int)
+			}
+			if settings["user_review_days"] != nil {
+				reviewDays = settings["user_review_days"].(int)
+			}
+		} else if strings.Contains(name, ":group.") || strings.HasPrefix(name, "unix.") {
+			memberType = "group"
+			if settings["group_expiry_days"] != nil {
+				expirationDays = settings["group_expiry_days"].(int)
+			}
+			if settings["group_review_days"] != nil {
+				reviewDays = settings["group_review_days"].(int)
+			}
+		} else {
+			memberType = "service"
+			if settings["service_expiry_days"] != nil {
+				expirationDays = settings["service_expiry_days"].(int)
+			}
+			if settings["service_review_days"] != nil {
+				reviewDays = settings["service_review_days"].(int)
+			}
+		}
+
+		if err := validateMemberReviewAndExpiration(data, expirationDays, reviewDays, memberType); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func validateMemberReviewAndExpiration(memberData map[string]interface{}, expirationDays int, reviewDays int, memberType string) error {
+	expiration := memberData["expiration"].(string)
+	review := memberData["review"].(string)
+
+	settingType := "expiration"
+	if err := validateMemberDate(expirationDays, expiration, memberType, settingType); err != nil {
+		return err
+	}
+
+	settingType = "review"
+	if err := validateMemberDate(reviewDays, review, memberType, settingType); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateMemberDate(days int, dateString string, memberType string, settingType string) error {
```

Comment:
please change memberType and settingType to enum

Reply:
I replaced the simple strings with string enums (basically consts)